File: backend/server/model/report.go
package model

import "time"

type ReportUser struct {
	ID          string    `gorm:"type:uuid;primary_key;default:uuid_generate_v4()" json:"id"`
	ReporterID  string    `gorm:"type:uuid" json:"reporter_id"`
	ReportedID  string    `gorm:"type:uuid" json:"reported_id"`
	Description string    `gorm:"type:text" json:"description"`
	CreatedAt   time.Time `json:"created_at"`

	Reporter *User `json:"reporter,omitempty"`
	Reported *User `json:"reported,omitempty"`
}

type ReportPost struct {
	ID          string `gorm:"type:uuid;primary_key;default:uuid_generate_v4()" json:"id"`
	ReporterID  string `gorm:"type:uuid" json:"reporter_id"`
	ReportedID  string `gorm:"type:uuid" json:"reported_id"`
	Description string `gorm:"type:text" json:"description"`

	Reporter  *User     `json:"reporter,omitempty"`
	Reported  *Post     `json:"reported,omitempty"`
	CreatedAt time.Time `json:"created_at"`
}

File: frontend/ReSell/app/src/main/java/com/example/resell/model/Report.kt
package com.example.resell.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import java.time.OffsetDateTime

@JsonClass(generateAdapter = true)
data class ReportUser(
    @Json(name = "id")
    val id: String,

    @Json(name = "reporter_id")
    val reporterId: String,

    @Json(name = "reported_id")
    val reportedId: String,

    @Json(name = "description")
    val description: String,

    @Json(name = "created_at")
    val createdAt: OffsetDateTime,

    @Json(name = "reporter")
    val reporter: User? = null,

    @Json(name = "reported")
    val reported: User? = null
)

@JsonClass(generateAdapter = true)
data class ReportPost(
    @Json(name = "id")
    val id: String,

    @Json(name = "reporter_id")
    val reporterId: String,

    @Json(name = "reported_id")
    val reportedId: String,

    @Json(name = "description")
    val description: String,

    @Json(name = "created_at")
    val createdAt: OffsetDateTime,

    @Json(name = "reporter")
    val reporter: User? = null,

    @Json(name = "reported")
    val reported: Post? = null
)
